src/arch/ia32/descriptor/tss.rs

```diff
@@ -1,65 +1,139 @@
-use super::segment::lower::Lower;
-use super::{split_address, split_limit, Granularity};
+//! A module containing the implementation, tests and types around the
+//! [`TssDescriptor`] structure to describe the permissions and capacities
+//! of a task state segment.
+use super::Granularity;
 use crate::arch::ia32::PrivilegeLevel;
-use upper::Upper;
+use bit_field::BitField;
+use configuration::Configuration;
+use permissions::Permissions;
 
-mod upper;
+mod configuration;
+mod permissions;
 
+/// A task state segment descriptor structure that can be used directly by the
+/// processor to describe a task state segment.
 #[must_use]
-#[derive(Copy, Clone)]
+#[derive(Default, Copy, Clone)]
 #[repr(C, packed)]
 pub struct TssDescriptor {
```

Comment:
Don't you want to implement `fmt::Display` for this structure ?
You can leave a `todo!` if you want as in the ia32e gate.rs.

Reply:
The current display format is not the final one, I'm not too sure about what we should do or what we should need. We'll keep the current code and implement proper display trait for those  structure in another PR.